lib/tokenize.cpp

```diff
@@ -7616,11 +7616,43 @@ void Tokenizer::findGarbageCode() const
         if (tok->str() == "{")
             tok = tok->link();
         else if (tok->isKeyword() && Token::simpleMatch(tok->previous(), ") try")) { // function-try-block
-            const Token* endTok = Token::findsimplematch(tok->next(), "catch (");
-            if (endTok)
-                endTok = endTok->next()->link();
-            if (Token::simpleMatch(endTok, ") {"))
-                tok = endTok->next()->link();
+            const Token* endTry = tok->next();
```

Comment:
It doesn't seem the code `.. ) try { ..` is handled well here. endTry will point at `{` and that does not match "%name% (|{" so the loop is skipped and then it seems there will be a syntax error because it's not `}`. I'd suggest something like:

    // skip initializer list:
    while (Token::Match(endTry, "[:,] %name% (|{"))
        endTry = endTry->linkAt(2)->next();
    if (!Token::simpleMatch(endTry, "{"))
        // some syntaxError
    // skip try body
    endTry = endTry->link();
    // skip catch..
    while (Token::simpleMatch(endTry, "} catch (") && Token::simpleMatch(endTry->linkAt(2), ") {"))
        endTry = endTry->linkAt(2)->linkAt(1);

Reply:
Thanks, I didn't think about an empty initializer list.
Turns out, we have `simplifyFunctionTryCatch()` already, which adds an extra scope beforehand. Should we remove that?

Reply:
As far as I see it's a good idea to add an extra scope. It doesn't change the semantics somehow does it? I'm not totally familiar with function-try-catch blocks.

> Otherwise, simplifyFunctionTryCatch() could be updated to also handle initializer lists.
:+1:

Reply:
If we did that (update simplifyFunctionTryCatch() to handle initializer lists), the present PR would become redundant, since the try/catch would no longer be in the global scope.